src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1735,12 +1735,7 @@ private static void GetStateMachineSlotDebugInfo(
 
             initializersBody ??= GetSynthesizedEmptyBody(method);
 
-            if (method is SynthesizedPrimaryConstructor primaryCtor && method.ContainingType.IsStructType())
-            {
-                body = BoundBlock.SynthesizedNoLocals(primaryCtor.GetSyntax());
```

Comment:
What replicates this logic now? What body is created for the primary constructor?

Reply:
The same logic which was used for class primary constructors is now also used for struct primary constructors. You can see that where the assertions were adjusted.

I reviewed the code around this area and didn't spot anything which would be problematic for structs, but, it's quite possible I missed something.